### ISSUE TITLE
no-handler check reported on a handler

### ISSUE THREAD

**Issue by TomasTomecek** [ansible/ansible-lint#3222]
##### Summary

```
no-handler: Tasks that run when changed should likely be handlers.
playbooks/deploy.yml:439 Task/Handler: Restart nginx deployment
```

That line is in a handler:
https://github.com/packit/deployment/blob/82754e7c527e7c0f9f20dbdcc75b2aa393eb3b25/playbooks/deploy.yml#L439-L443

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

via pre-commit:
```yaml
repos:
  - repo: https://github.com/ansible/ansible-lint.git
    rev: v6.14.3
```

##### Desired Behavior

Don't raise no-handler warnings for handlers.

**Comment by ajinkyau** [comment]
@TomasTomecek Handlers are basically tasks so IMO it's not false positives.
Also, if you use `changed_when` the warnings will go away.

**Comment by thetic** [comment]
> Handlers are basically tasks so IMO it's not false positives.

Strongly disagree.

**Comment by ssbarnea** [comment]
@TomasTomecek @thetic  We just released https://github.com/ansible/ansible-lint/releases/tag/v6.14.4 which includes two changes related to your bug report.

We ensured that `when: not foo.changed` does not trigger no-handler rule and we also added some documentation explaining why you still need to add a `changed_when` for handlers that run external commands, basically because they are also tasks that can be chained.
